src/io_device_traverse.rs

```diff
@@ -170,7 +170,10 @@ extern "C" fn inner_traverse_channel<ChannelData, Ctx>(
     let ctx = TraverseCtx::<ChannelData, Ctx>::from_iter(i);
     let mut chan = IoChannel::<ChannelData>::from_iter(i);
 
-    let rc = (ctx.channel_cb)(chan.channel_data_mut(), &mut ctx.ctx);
+    let mut rc = ChannelTraverseStatus::Ok;
+    if chan.thread_name() != "init_thread" {
```

Comment:
This might work for now but seems quite brittle.. we probably better do this properly by using channel specific data. Example: Adding bool io_channel to the NexusChannel perhaps

Reply:
Alternatively, I'll let this closure call happen irrespective of spdk thread and read the new boolean in the nexus code just before callling chan.reconnect_all()

Reply:
or better, can get the id of current thread by spdk_thread_get_id() using `spdk_rs::Thread` abstraction of spdk_thread, and if id is `1` it's `init_thread`.

Reply:
Deferring the decision to the "rust native" code seems good.